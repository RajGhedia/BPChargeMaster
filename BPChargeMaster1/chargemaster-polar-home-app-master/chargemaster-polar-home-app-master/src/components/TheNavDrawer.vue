<template>
    <v-navigation-drawer v-model="open" absolute temporary>
        <v-layout>
            <v-flex xs7 offset-xs1 pt-3 pb-3>
                <v-img :src="require('@/assets/logo.svg')" contain></v-img>
            </v-flex>
        </v-layout>
        <v-divider></v-divider>
        <v-list two-line>
            <template v-for="(item, index) in items">
                <v-list-tile :key="item.icon" ripple :to="item.to">
                    <v-list-tile-avatar>
                        <v-img
                            :src="require(`@/assets/${item.icon}.svg`)"
                            class="icon"
                        ></v-img>
                    </v-list-tile-avatar>

                    <v-list-tile-content>
                        <v-list-tile-title class="primary--text">{{
                            item.title
                        }}</v-list-tile-title>
                    </v-list-tile-content>

                    <v-list-tile-action>
                        <v-btn icon ripple :to="item.to">
                            <v-icon color="primary">chevron_right</v-icon>
                        </v-btn>
                    </v-list-tile-action>
                </v-list-tile>
                <v-divider :key="index"></v-divider>
            </template>
        </v-list>
    </v-navigation-drawer>
</template>
<script>
import EventBus from '@/event-bus'
export default {
    methods: {
        openDrawer() {
            this.open = true
        }
    },
    data() {
        return {
            open: false,
            items: [
                { icon: 'info', title: "FAQ's", to: '/faqs' },
                { icon: 'phone', title: 'Contact Us', to: '/contact' },
                { icon: 'polar-icon', title: 'Polar Plus Network' },
                { icon: 'tc', title: "Privacy Policy, T&C's", to: '/privacy' }
            ]
        }
    },
    created() {
        EventBus.$on('openDrawer', this.openDrawer)
    },
    beforeDestroy() {
        EventBus.$off('openDrawer', this.openDrawer)
    }
}
</script>
