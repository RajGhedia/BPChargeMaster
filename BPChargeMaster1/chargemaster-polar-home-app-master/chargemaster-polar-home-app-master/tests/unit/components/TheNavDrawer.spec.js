import TheNavDrawer from '@/components/TheNavDrawer.vue'
import Vue from 'vue'
import Vuetify from 'vuetify'
import { mount, createLocalVue, RouterLinkStub } from '@vue/test-utils'
import EventBus from '@/event-bus'

const localVue = createLocalVue()
Vue.use(Vuetify)

describe('TheNavDrawer.vue', () => {
    it('renders empty by default', () => {
        const wrapper = getWrapper()

        expect(wrapper.element).toMatchSnapshot()
    })

    it('listens for the notify event', () => {
        const wrapper = getWrapper()

        EventBus.$emit('openDrawer')

        expect(wrapper.vm.open).toBe(true)
        expect(wrapper.element).toMatchSnapshot()
    })

    it('cleans up event listener when component is destroyed', () => {
        const wrapper = getWrapper()

        wrapper.setMethods({
            openDrawer: jest.fn()
        })

        const openDrawer = wrapper.vm.openDrawer
        const cleanup = jest.fn()

        EventBus.$off = cleanup
        wrapper.destroy()

        expect(cleanup).toBeCalledWith('openDrawer', openDrawer)
        EventBus.$emit('openDrawer')
        expect(openDrawer).not.toHaveBeenCalled()
    })
})

function getWrapper() {
    return mount(TheNavDrawer, {
        localVue,
        stubs: {
            'router-link': RouterLinkStub
        }
    })
}
