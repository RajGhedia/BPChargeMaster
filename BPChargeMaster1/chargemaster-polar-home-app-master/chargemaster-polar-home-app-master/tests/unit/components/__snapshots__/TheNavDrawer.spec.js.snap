// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheNavDrawer.vue listens for the notify event 1`] = `
<aside
  class="v-navigation-drawer v-navigation-drawer--absolute v-navigation-drawer--open v-navigation-drawer--temporary theme--light"
  style="height: 100%; margin-top: 0px; transform: translateX(0px); width: 300px;"
>
  <div
    class="layout"
  >
    <div
      class="flex xs7 offset-xs1 pt-3 pb-3"
    >
      <div
        class="v-responsive v-image"
      >
        <div
          class="v-responsive__content"
        />
      </div>
    </div>
  </div>
   
  <hr
    class="v-divider theme--light"
  />
   
  <div
    class="v-list v-list--two-line theme--light"
    role="list"
  >
    <div
      role="listitem"
    >
      <a
        class="v-list__tile v-list__tile--link theme--light"
      >
        <div
          class="v-list__tile__avatar"
        >
          <div
            class="v-avatar"
            style="height: 40px; width: 40px;"
          >
            <div
              class="v-responsive v-image icon"
            >
              <div
                class="v-responsive__content"
              />
            </div>
          </div>
        </div>
         
        <div
          class="v-list__tile__content"
        >
          <div
            class="v-list__tile__title primary--text"
          >
            FAQ's
          </div>
        </div>
         
        <div
          class="v-list__tile__action"
        >
          <a
            class="v-btn v-btn--icon v-btn--router theme--light"
          >
            <div
              class="v-btn__content"
            >
              <i
                aria-hidden="true"
                class="v-icon material-icons theme--light primary--text"
              >
                chevron_right
              </i>
            </div>
          </a>
        </div>
      </a>
    </div>
     
    <hr
      class="v-divider theme--light"
    />
    <div
      role="listitem"
    >
      <a
        class="v-list__tile v-list__tile--link theme--light"
      >
        <div
          class="v-list__tile__avatar"
        >
          <div
            class="v-avatar"
            style="height: 40px; width: 40px;"
          >
            <div
              class="v-responsive v-image icon"
            >
              <div
                class="v-responsive__content"
              />
            </div>
          </div>
        </div>
         
        <div
          class="v-list__tile__content"
        >
          <div
            class="v-list__tile__title primary--text"
          >
            Contact Us
          </div>
        </div>
         
        <div
          class="v-list__tile__action"
        >
          <a
            class="v-btn v-btn--icon v-btn--router theme--light"
          >
            <div
              class="v-btn__content"
            >
              <i
                aria-hidden="true"
                class="v-icon material-icons theme--light primary--text"
              >
                chevron_right
              </i>
            </div>
          </a>
        </div>
      </a>
    </div>
     
    <hr
      class="v-divider theme--light"
    />
    <div
      role="listitem"
    >
      <div
        class="v-list__tile theme--light"
      >
        <div
          class="v-list__tile__avatar"
        >
          <div
            class="v-avatar"
            style="height: 40px; width: 40px;"
          >
            <div
              class="v-responsive v-image icon"
            >
              <div
                class="v-responsive__content"
              />
            </div>
          </div>
        </div>
         
        <div
          class="v-list__tile__content"
        >
          <div
            class="v-list__tile__title primary--text"
          >
            Polar Plus Network
          </div>
        </div>
         
        <div
          class="v-list__tile__action"
        >
          <button
            class="v-btn v-btn--icon theme--light"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              <i
                aria-hidden="true"
                class="v-icon material-icons theme--light primary--text"
              >
                chevron_right
              </i>
            </div>
          </button>
        </div>
      </div>
    </div>
     
    <hr
      class="v-divider theme--light"
    />
    <div
      role="listitem"
    >
      <a
        class="v-list__tile v-list__tile--link theme--light"
      >
        <div
          class="v-list__tile__avatar"
        >
          <div
            class="v-avatar"
            style="height: 40px; width: 40px;"
          >
            <div
              class="v-responsive v-image icon"
            >
              <div
                class="v-responsive__content"
              />
            </div>
          </div>
        </div>
         
        <div
          class="v-list__tile__content"
        >
          <div
            class="v-list__tile__title primary--text"
          >
            Privacy Policy, T&C's
          </div>
        </div>
         
        <div
          class="v-list__tile__action"
        >
          <a
            class="v-btn v-btn--icon v-btn--router theme--light"
          >
            <div
              class="v-btn__content"
            >
              <i
                aria-hidden="true"
                class="v-icon material-icons theme--light primary--text"
              >
                chevron_right
              </i>
            </div>
          </a>
        </div>
      </a>
    </div>
     
    <hr
      class="v-divider theme--light"
    />
  </div>
  <div
    class="v-navigation-drawer__border"
  />
</aside>
`;

exports[`TheNavDrawer.vue renders empty by default 1`] = `
<aside
  class="v-navigation-drawer v-navigation-drawer--absolute v-navigation-drawer--close v-navigation-drawer--temporary theme--light"
  style="height: 100%; margin-top: 0px; transform: translateX(-300px); width: 300px;"
>
  <div
    class="layout"
  >
    <div
      class="flex xs7 offset-xs1 pt-3 pb-3"
    >
      <div
        class="v-responsive v-image"
      >
        <div
          class="v-responsive__content"
        />
      </div>
    </div>
  </div>
   
  <hr
    class="v-divider theme--light"
  />
   
  <div
    class="v-list v-list--two-line theme--light"
    role="list"
  >
    <div
      role="listitem"
    >
      <a
        class="v-list__tile v-list__tile--link theme--light"
      >
        <div
          class="v-list__tile__avatar"
        >
          <div
            class="v-avatar"
            style="height: 40px; width: 40px;"
          >
            <div
              class="v-responsive v-image icon"
            >
              <div
                class="v-responsive__content"
              />
            </div>
          </div>
        </div>
         
        <div
          class="v-list__tile__content"
        >
          <div
            class="v-list__tile__title primary--text"
          >
            FAQ's
          </div>
        </div>
         
        <div
          class="v-list__tile__action"
        >
          <a
            class="v-btn v-btn--icon v-btn--router theme--light"
          >
            <div
              class="v-btn__content"
            >
              <i
                aria-hidden="true"
                class="v-icon material-icons theme--light primary--text"
              >
                chevron_right
              </i>
            </div>
          </a>
        </div>
      </a>
    </div>
     
    <hr
      class="v-divider theme--light"
    />
    <div
      role="listitem"
    >
      <a
        class="v-list__tile v-list__tile--link theme--light"
      >
        <div
          class="v-list__tile__avatar"
        >
          <div
            class="v-avatar"
            style="height: 40px; width: 40px;"
          >
            <div
              class="v-responsive v-image icon"
            >
              <div
                class="v-responsive__content"
              />
            </div>
          </div>
        </div>
         
        <div
          class="v-list__tile__content"
        >
          <div
            class="v-list__tile__title primary--text"
          >
            Contact Us
          </div>
        </div>
         
        <div
          class="v-list__tile__action"
        >
          <a
            class="v-btn v-btn--icon v-btn--router theme--light"
          >
            <div
              class="v-btn__content"
            >
              <i
                aria-hidden="true"
                class="v-icon material-icons theme--light primary--text"
              >
                chevron_right
              </i>
            </div>
          </a>
        </div>
      </a>
    </div>
     
    <hr
      class="v-divider theme--light"
    />
    <div
      role="listitem"
    >
      <div
        class="v-list__tile theme--light"
      >
        <div
          class="v-list__tile__avatar"
        >
          <div
            class="v-avatar"
            style="height: 40px; width: 40px;"
          >
            <div
              class="v-responsive v-image icon"
            >
              <div
                class="v-responsive__content"
              />
            </div>
          </div>
        </div>
         
        <div
          class="v-list__tile__content"
        >
          <div
            class="v-list__tile__title primary--text"
          >
            Polar Plus Network
          </div>
        </div>
         
        <div
          class="v-list__tile__action"
        >
          <button
            class="v-btn v-btn--icon theme--light"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              <i
                aria-hidden="true"
                class="v-icon material-icons theme--light primary--text"
              >
                chevron_right
              </i>
            </div>
          </button>
        </div>
      </div>
    </div>
     
    <hr
      class="v-divider theme--light"
    />
    <div
      role="listitem"
    >
      <a
        class="v-list__tile v-list__tile--link theme--light"
      >
        <div
          class="v-list__tile__avatar"
        >
          <div
            class="v-avatar"
            style="height: 40px; width: 40px;"
          >
            <div
              class="v-responsive v-image icon"
            >
              <div
                class="v-responsive__content"
              />
            </div>
          </div>
        </div>
         
        <div
          class="v-list__tile__content"
        >
          <div
            class="v-list__tile__title primary--text"
          >
            Privacy Policy, T&C's
          </div>
        </div>
         
        <div
          class="v-list__tile__action"
        >
          <a
            class="v-btn v-btn--icon v-btn--router theme--light"
          >
            <div
              class="v-btn__content"
            >
              <i
                aria-hidden="true"
                class="v-icon material-icons theme--light primary--text"
              >
                chevron_right
              </i>
            </div>
          </a>
        </div>
      </a>
    </div>
     
    <hr
      class="v-divider theme--light"
    />
  </div>
  <div
    class="v-navigation-drawer__border"
  />
</aside>
`;
